### DISCUSSION TITLE
UI enhancements

### DISCUSSION THREAD

**Discussion by moalamri** [directus/directus#12475] [Feature Requests]
I love the UI system used in Directus.
Everything is perfect except some few things that could be enhanced (in my opinion). I'm not a UI/UX expert, so I'm going to need your opinion about this.
1- Custom scrollbar
2- Override browser's input autofill highlight style.

Here are some before/after examples of how it would look if applied.

### Scrollbar (before)
![image](https://user-images.githubusercontent.com/964077/160749651-ef6ed17a-d609-4989-9f58-4e26847e2881.png)

![image](https://user-images.githubusercontent.com/964077/160749716-a18624e7-54b4-4c31-904f-772ce4f1ebf8.png)

### Scrollbar (after)
![image](https://user-images.githubusercontent.com/964077/160749683-81a2995d-382f-4753-95b8-67acc8c48e00.png)

![image](https://user-images.githubusercontent.com/964077/160749741-7cee91be-4233-4a07-8c59-df1b4e39fa1c.png)

### Input autofill highlight (before)
![image](https://user-images.githubusercontent.com/964077/160749860-1923750b-89df-4e2a-9dcc-0ae69e105c23.png)

### Input autofill highlight (after)
![image](https://user-images.githubusercontent.com/964077/160749888-52227bb7-b238-4d87-9ecc-474bbd0f45c1.png)

What do you think?

**Comment by azrikahar** [comment]
I'm not a UI/UX expert either, but I've also thought about the scrollbar change before 😄 I believe we should wait for the new [CSS Scrollbars](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars) as a better and standard approach 👍

The `::-webkit-scrollbar` CSS pseudo element is deemed as [non-standard in MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar), and it seems the [CSS Scrollbar spec does note the following statement](https://drafts.csswg.org/css-scrollbars/#out-of-scope):

> Exposing the scrollbar-related `::-webkit-` prefixed pseudo-elements to the Web is considered a mistake by both the CSS Working Group and Webkit.

---

As for autofill, I do agree that it'd be nice if we can somehow make it more inline with the current styles. I'm not sure would it be okay to flat out remove any styling though, as it may be seen as "breaking" a known experience (as well as not being as obvious which ones are the autofilled fields now).

**Reply by benhaynes** [reply]
Scroll bars should be done with proper CSS, which we can add now ([prefixed](https://caniuse.com/css-scrollbar)) in preparation for full support.

As for autofill styling, we can update to `--primary-subdued` so it matches color, and then try to fix the padding issues that make it appear cut-off. Below are two options... but I think the background option is more standard/recognizable.
![Screen Shot 2022-03-30 at 9 50 27 AM](https://user-images.githubusercontent.com/522079/160850518-2781a57b-75ac-4aa8-8ec1-e30d5dcf1dd7.png)

**Reply by moalamri** [reply]
Hi @benhaynes @azrikahar Thanks for sharing your thoughts and ideas about this. It’s good to know they will be planned to be done in the proper way.
Can’t wait to have them implemented :)
As for autofill styling, I vote for the background one although It could be disturbing to have when the autofilled inputs are in random locations.

**Reply by benhaynes** [reply]
> It could be disturbing to have when the autofilled inputs are in random locations.

True, but the point is to differentiate which inputs have been autofilled, so I think that's expected/ok. 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
